### PR TITLE
[Pyamqp] Pyampq debug build Linkedin

### PR DESCRIPTION
**Pull request by kashifkhan**
This PR brings in the logging added in from Anna's PR & Swathis previous wheel. The bug fixes around auth_status and network trace are already in pyproto/main via the following commits

- [#25295 ](https://github.com/Azure/azure-sdk-for-python/pull/25218)
- [#25024](https://github.com/Azure/azure-sdk-for-python/pull/25024)

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py [hunk 1]
```diff
@@ -311,7 +312,9 @@ def send_batch(self, event_data_batch, **kwargs):
             cast(EventHubProducer, self._producers[partition_id]).send(
                 to_send_batch, timeout=send_timeout
             )
-        except (KeyError, AttributeError, EventHubError):
+        except (KeyError, AttributeError, EventHubError) as e:
+            _LOGGER.info(
+                "Producer for partition ID '{}' not available: {}. Rebuilding new producer.".format(partition_id, e))
```
Comment (l0lawrence) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py:317]: would this exception populate anything related to the token -- should it be debug level (pylint checker issue related)?

File: sdk/eventhub/azure-eventhub/samples/sync_samples/send.py [hunk 2]
```diff
@@ -9,15 +9,18 @@
 Examples to show sending events with different options to an Event Hub partition.
 """
 
+import logging
+import sys
 import time
 import os
+from tkinter.tix import Tree
```
Comment (l0lawrence) [inline sdk/eventhub/azure-eventhub/samples/sync_samples/send.py:16]: I don't know if this is used

**Review by l0lawrence — COMMENTED** [review]
Lgtm I just had 2 small questions

File: sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py (hunk 1 above)
Comment (kashifkhan) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py:317]: The idea for this one is to figure out why are we going towards `start_producer` in the first place ( keyerror, object is None, or some other underlying issue that comes in ). 

Its after this point when the management request happens and there if/elif/else block that Ive added in that captures if the token is empty and if not print out the expires_on time.

**Review by l0lawrence — APPROVED** [review]

File: sdk/eventhub/azure-eventhub/CHANGELOG.md [hunk 3]
```diff
@@ -1,15 +1,15 @@
 # Release History
 
-## 5.8.0b5 (Unreleased)
-
-### Features Added
-
-### Breaking Changes
+## 5.8.0b5 (2022-07-19)
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/CHANGELOG.md:3]: a5? also in _version.py

File: sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py [hunk 4]
```diff
@@ -386,7 +386,19 @@ def _management_request(self, mgmt_msg, op_type):
...
+                access_token = mgmt_auth.get_token()
+
+                if not access_token:
+                    _LOGGER.info("Management client received an access token object")
+
+                elif not access_token.token:
+                    _LOGGER.info("Management client received an empty token")
+
+                else:
+                    _LOGGER.info(f"Management client token expires on: {datetime.fromtimestamp(access_token.expires_on)}")
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py:398]: just checking, but we want all of these to be info and not debug?

File: sdk/eventhub/azure-eventhub/samples/sync_samples/send.py [hunk 5]
```diff
@@ -96,4 +96,4 @@ def send_event_data_list(producer):
     send_event_data_batch_with_properties(producer)
     send_event_data_list(producer)
 
-print("Send messages in {} seconds.".format(time.time() - start_time))
+print("Send messages in {} seconds.".format(time.time() - start_time))
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/samples/sync_samples/send.py:99]: looks like no changes were made to this sample, so you can probably revert it? I think the CI pylint might fail since this file doesn't have a newline at the end.

File: sdk/eventhub/azure-eventhub/CHANGELOG.md [hunk 6]
```diff
@@ -1,15 +1,15 @@
...
 
 ### Bugs Fixed
 
+- Bug fixes that would prevent token refresh at regular intervals
+- Pass in the proper kwarg, so that debug levels will be output when requested
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/CHANGELOG.md:8]: for conformity with rest of changelog, maybe rewording like this?
```suggestion
- Fixed bug that prevented token refresh at regular intervals.
- Fixed bug that was improperly passing the debug keyword argument, so that network trace debug logs are output when requested.
```
for second bullet, I think this was the network_trace kwarg, right?

File: sdk/eventhub/azure-eventhub/CHANGELOG.md [hunk 7]
```diff
@@ -1,15 +1,15 @@
...
 ### Other Changes
 
+- Specific logging added in to track error scenarios
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/CHANGELOG.md:11]:
```suggestion
- Added logging to track error scenarios.
```
+ for our future selves, maybe more details on which error scenarios?

**Review by swathipil — APPROVED** [review]
logging lgtm! left a few other comments. (+ you might plan on doing this already, but just a reminder to run the PR against the test pipeline :) )

File: sdk/eventhub/azure-eventhub/CHANGELOG.md (hunk 6 above)
Comment (kashifkhan) [reply sdk/eventhub/azure-eventhub/CHANGELOG.md:8]: right it was the network_trace

File: sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py [hunk 8]
```diff
@@ -386,7 +386,19 @@ def _management_request(self, mgmt_msg, op_type):
                 mgmt_client.open()
                 while not mgmt_client.client_ready():
                     time.sleep(0.05)
-                mgmt_msg.application_properties["security_token"] = mgmt_auth.get_token()
+                access_token = mgmt_auth.get_token()
+
+                if not access_token:
+                    _LOGGER.info("Management client received an access token object")
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py:392]: Shouldn't this be something like "Management client received an **empty** access token object"?
